derive-error-chain/src/lib.rs

```diff
@@ -258,6 +258,37 @@ pub fn derive_error_chain(input: proc_macro::TokenStream) -> proc_macro::TokenSt
 	let ast = syn::parse_derive_input(&source).unwrap();
 	let error_kind_name = ast.ident;
 
+	let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
+	let mut generics_lifetime = ast.generics.clone();
+	generics_lifetime.lifetimes.push(syn::LifetimeDef::new("'error_chain_lifetime"));
+	let (impl_generics_lifetime, _, _) = generics_lifetime.split_for_impl();
+	let mut generics_ext1 = ast.generics.clone();
+	generics_ext1.ty_params.push(syn::TyParam {
+		attrs: Vec::new(),
+		ident: syn::Ident::from("DERIVE_ERROR_CHAIN_T"),
+		bounds: Vec::new(),
+		default: None,
+	});
+	let (_, ty_generics_ext1, _) = generics_ext1.split_for_impl();
+	let mut generics_ext2 = ast.generics.clone();
+	generics_ext2.ty_params.push(syn::TyParam {
+		attrs: Vec::new(),
+		ident: syn::Ident::from("DERIVE_ERROR_CHAIN_T"),
```

Comment:
I don't know of any precedent in custom derives about naming generics that don't clash with generics in the input, but perhaps `__T`, `__E`, `__F`, `__K` would be better?